src/extensions/CanvasWidget/CanvasWidget.tsx

```diff
@@ -89,25 +89,36 @@ const CanvasWidget = (props: Props) => {
 
     // To learn about callback refs: https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node
     const [divElement, setDivElement] = useState<HTMLDivElement | null>(null)
+    const [hasFocus, setHasFocus] = useState<boolean>(false)
     const divRef = React.useCallback((elmt: HTMLDivElement) => {
         // this should get called only once after the div has been written to the DOM
         // we set this div element so that it can be used below when we set the canvas
         // elements to the layers
         setDivElement(elmt)
     }, [])
 
+    useEffect(() => {
+        if (divElement) {
+            (divElement as any)['_hasFocus'] = hasFocus
```

Comment:
The mouse wheel event does not have access to the state. However, it does have access to the divElement. Therefore we can attach something to the divElement that can be read on the mouse wheel event to determine whether we preventDefault. This could also be done using useRef(). The latter would probably be better as we would not need to hack the (divElement as any).